### PR TITLE
[ST] Parallelize kubeClient interactions in AbstractUpgradeST

### PR DESCRIPTION
**Pull request by lhunyady**
### Type of change
- Enhancement 
- Refactoring

### Description

Parallelize kubeClient interactions in AbstractUpgradeSt

### CODE REVIEW

**Comment by lhunyady** [comment]
I have not ran extensive performance tests on this, but with this small change locally I was able to save 54 seconds out of 13 min 46 sec or 7% run time just on the StrimziUpgradeST::testUpgradeKafkaWithoutVersion alone.
 
Screenshot 2024-08-29 at 08 48 19
Screenshot 2024-08-29 at 11 39 48

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java [hunk 1]
```diff
@@ -222,16 +224,14 @@ protected void logPodImagesWithConnect(String namespaceName) {
     }
 
     protected void logPodImages(String namespaceName, LabelSelector... labelSelectors) {
-        for (LabelSelector labelSelector : labelSelectors) {
-            List<Pod> pods = kubeClient().listPods(namespaceName, labelSelector);
-
-            pods.forEach(pod ->
+        Arrays.stream(labelSelectors).parallel()
+            .map(selector -> kubeClient().listPods(namespaceName, selector))
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java:228]:
```suggestion
        Arrays.stream(labelSelectors)
            .parallel()
            .map(selector -> kubeClient().listPods(namespaceName, selector))
```

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java [hunk 2]
```diff
@@ -355,9 +355,13 @@ protected void setupEnvAndUpgradeClusterOperator(BundleVersionModificationData u
...
-            for (int x = 0; x < upgradeTopicCount; x++) {
-                cmdKubeClient().applyContent(TestUtils.getContent(kafkaTopicYaml, TestUtils::toYamlString).replace("name: \"my-topic\"", "name: \"" + topicName + "-" + x + "\""));
-            }
+
+            String topicNameTemplate = topicName + "-%s";
+            IntStream.range(0, upgradeTopicCount)
+                    .mapToObj(topicNameTemplate::formatted)
+                    .map(this::getKafkaYamlWithName)
+                    .parallel()
+                    .forEach(cmdKubeClient()::applyContent);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java:364]:
```suggestion
                .mapToObj(topicNameTemplate::formatted)
                .map(this::getKafkaYamlWithName)
                .parallel()
                .forEach(cmdKubeClient()::applyContent);
```

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java [hunk 3]
```diff
@@ -398,6 +402,14 @@ protected void setupEnvAndUpgradeClusterOperator(BundleVersionModificationData u
         makeSnapshots();
     }
 
+    private String getKafkaYamlWithName(String name) {
+        String initialName = "name: \"my-topic\"";
+        String newName =  "name: \"%s\"".formatted(name);
+
+        return TestUtils.getContent(kafkaTopicYaml, TestUtils::toYamlString)
+                .replace(initialName, newName);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java:410]:
```suggestion
        return TestUtils.getContent(kafkaTopicYaml, TestUtils::toYamlString).replace(initialName, newName);
```

**Review by im-konge — COMMENTED** [review]
Just 3 small nits regarding code style, but otherwise this LGTM. Thanks for the PR! :)

**Comment by lhunyady** [comment]
Thank you! I have applied them!

**Comment by lhunyady** [comment]
Not sure what these errors are about:
https://dev.azure.com/cncf/strimzi/_build/results?buildId=178807&view=logs&j=793c1cfc-54ce-5b9b-bdf3-ecb49b674eb9&t=d0bfec6c-2454-5c05-1a4d-2be757165bc4&l=143

**Comment by Frawless** [comment]
@lhunyady there is ongoing minikube registry outage, the failure is not related to your changes.

**Comment by lhunyady** [comment]
Thx, is there a tracking ticket for that?

**Comment by Frawless** [comment]
There are several issues opened on Minikube gh. I opened https://github.com/strimzi/strimzi-kafka-operator/pull/10519 that should workaround the problem 🤞

**Review by henryZrncik — APPROVED** [review]

**Review by im-konge — APPROVED** [review]

**Comment by im-konge** [comment]
LGTM, thanks a lot for the PR @lhunyady !

**Comment by lhunyady** [comment]
Thank you @im-konge! Not sure what the next steps are, but from my point of view this PR can be merged!

**Comment by scholzj** [comment]
@lhunyady PRs require a review from 2 maintainers to be merged. I think that @see-quick or @Frawless will look at it next. But in any case, there is nothing you have to do right now.

**Comment by lhunyady** [comment]
Thank you everyone for the help!

**Pull request merged**